File: sdkwork-ai-sdk-swift/Sources/API/MessageApi.swift
import Foundation

public class MessageApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Count Claude tokens
    public func countClaudeTokens(body: CountClaudeTokensPostRequest) async throws -> CountClaudeTokensPostResponse? {
        let response = try await client.post(ApiPaths.aiPath("/messages/count_tokens"), body: body)
        return response as? CountClaudeTokensPostResponse
    }

    /// Create Claude message
    public func createClaude(body: CreateClaudeMessageRequest, headers: [String: String]? = nil) async throws -> CreateClaudeMessageResponse? {
        let response = try await client.post(ApiPaths.aiPath("/messages"), body: body, params: nil, headers: headers)
        return response as? CreateClaudeMessageResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/MessageApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class MessageApi
    {
        private readonly HttpClient _client;

        public MessageApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Count Claude tokens
        /// </summary>
        public async Task<CountClaudeTokensPostResponse?> CountClaudeTokensAsync(CountClaudeTokensPostRequest body)
        {
            return await _client.PostAsync<CountClaudeTokensPostResponse>(ApiPaths.AiPath("/messages/count_tokens"), body);
        }

        /// <summary>
        /// Create Claude message
        /// </summary>
        public async Task<CreateClaudeMessageResponse?> CreateClaudeAsync(CreateClaudeMessageRequest body, Dictionary<string, string>? headers = null)
        {
            return await _client.PostAsync<CreateClaudeMessageResponse>(ApiPaths.AiPath("/messages"), body, null, headers);
        }
    }
}
